Cabal/Distribution/Simple/Program/Ar.hs

```diff
@@ -97,9 +119,26 @@ createArLibArchive verbosity lbi targetPath files = do
   where
     progDb = withPrograms lbi
     Platform hostArch hostOS = hostPlatform lbi
-    verbosityOpts v | v >= deafening = ["-v"]
-                    | v >= verbose   = []
-                    | otherwise      = ["-c"]
+    verbosityOpts v
+      | v >= deafening = ["-v"]
+      | v >= verbose   = []
+      | otherwise      = ["-c"] -- Do not warn if library had to be created.
+
+-- | The @ar@ tool expects response file to contain sequence of strings
+-- delimited by whitespace. Thus, in order to handle file names with spaces
+-- they should be enclosed in single or double quotes.
+--
+-- Windows poses additional challenge to creating correct response files.
+-- Namely, on Windows standard path separator is backspace but in @ar@
+-- response file format it's escape sequence. Therefore backslashes must
+-- be escaped as well.
+escapeFileName :: FilePath -> FilePath
+escapeFileName = concatMap escapeChar
+  where
+    escapeChar :: Char -> String
+    escapeChar '"'  = "\\\""
+    escapeChar '\\' = "\\\\"
+    escapeChar c    = [c]
```

Comment:
I think it would be best to unify the two under single implementation in `Distribution.Simple.Program.ResponseFile`. Haddock's module has following comment

```
      -- Support a gcc-like response file syntax.  Each separate
      -- argument and its possible parameter(s), will be separated in the
      -- response file by an actual newline; all other whitespace,
      -- single quotes, double quotes, and the character used for escaping
      -- (backslash) are escaped.  The called program will need to do a similar
      -- inverse operation to de-escape and re-constitute the argument list.
```

which implies that it tries to match gcc escaping rules and that's precisely what we're trying to achieve for `ar`, and possibly `ld`.